tests/EnumDynamicReturnTypeExtensionTest.php

```diff
@@ -0,0 +1,33 @@
+<?php
+
+declare(strict_types=1);
+
+namespace MabeEnumPHPStanTest;
+
+use MabeEnum\Enum;
+use MabeEnumPHPStan\EnumDynamicReturnTypeExtension;
+
+final class EnumDynamicReturnTypeExtensionTest extends ExtensionTestCase
+{
+    /** @var EnumDynamicReturnTypeExtension */
+    private $extension;
+
+    protected function setUp(): void
+    {
+        parent::setUp();
+
+        $this->extension = new EnumDynamicReturnTypeExtension();
+    }
+
+    public function testUnsupportedMethod(): void
+    {
+        $code = <<<'CODE'
+<?php
+function f(MabeEnumPHPStanTest\Assets\IntTypeEnum $enum) {
+    die;
+}
+CODE;
+
+        $this->processCode($code, '$enum->getName()', 'string', $this->extension);
```

Comment:
btw we can support this too 😄

Reply:
One after the next.

The return type is already known. We could just make it to return `ConstantStringType` instead.

A more interesting feature would be to detect which enumerator has been instantiated via `MyEnum::ENUMERATOR()` to then know the exact type and value.

```php
class MyEnum extends Enum {
    const INT = 1;
    const STR = 'str';
}

$int = MyEnum::INT(); // detect $int refers to constant `INT`
$int->getValue(); // detect this returns 1 instead of 1|'str'
```

Reply:
yeah, i was thinking about the same

i tried generics, but it returns generalized type :(

https://phpstan.org/r/8dd182c6-7bfd-4c09-9337-e25f1c6a29d2